### PR TITLE
MOD: add payout_base,min,max params in function ad::create

### PR DESCRIPTION
**Pull request by RightFS**
we need update our apps to fit the changes of this pr @hikaruchang 

**Type of change**

* [ ] Bug Fix (non-breaking change which fixes an issue)
* [x] New Feature (non-breaking change which adds functionality)
* [x] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update
* [ ] This change requires a storage migration
* [ ] This change requires a benchmark

**Checklist**

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] New and existing benchmarks pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream modules
* [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pallets/ad/src/lib.rs [hunk 1]
```diff
@@ -261,6 +262,9 @@ pub mod pallet {
                     metadata,
                     reward_rate,
                     created,
+                    payout_base,
+                    payout_min,
+                    payout_max,
```
Comment (coderfengyun) [inline pallets/ad/src/lib.rs:267]: 针对已经存在的ad需要做一下storage migration.

File: pallets/ad/src/tests.rs [hunk 2]
```diff
@@ -59,7 +62,17 @@ fn should_create() {
 fn should_fail_when_insufficient() {
```
Comment (coderfengyun) [inline pallets/ad/src/tests.rs:62]: 测试用例里应当补充小于min和大于max的场景；以及payoutbase的配置值需要覆盖几个边界值如极小，极大。

File: pallets/ad/src/lib.rs (hunk 1 above)
Comment (RightFS) [reply pallets/ad/src/lib.rs:267]: done

File: pallets/ad/src/tests.rs (hunk 2 above)
Comment (RightFS) [reply pallets/ad/src/tests.rs:62]: done

File: pallets/ad/Cargo.toml [hunk 3]
```diff
@@ -40,7 +40,7 @@ sp-std = { git = 'https://github.com/paritytech/substrate', branch = 'polkadot-v
 frame-benchmarking = { git = 'https://github.com/paritytech/substrate', default-features = false, optional = true, branch = 'polkadot-v0.9.13' }
 frame-support = { git = 'https://github.com/paritytech/substrate', branch = 'polkadot-v0.9.13', default-features = false }
 frame-system = { git = 'https://github.com/paritytech/substrate', branch = 'polkadot-v0.9.13', default-features = false }
-
+parami-primitives = { path = '../../primitives', default-features = false }
```
Comment (coderfengyun) [inline pallets/ad/Cargo.toml:43]: 可以放在dev-dependencies里

File: pallets/ad/src/lib.rs [hunk 4]
```diff
@@ -181,6 +185,34 @@ pub mod pallet {
...
+                    "migration error"
+                );
+                assert_eq!(
+                    ad_meta.payout_max,
+                    (10 * DOLLARS).saturated_into(),
+                    "migration error"
+                );
+            }
+
+            Ok(())
```
Comment (coderfengyun) [inline pallets/ad/src/lib.rs:214]: 这里的逻辑应该是：只有在version是2的时候才执行，且非version2是不能报错。

File: pallets/ad/Cargo.toml (hunk 3 above)
Comment (RightFS) [reply pallets/ad/Cargo.toml:43]: 为什么它能放到dev-dependencies里

File: pallets/ad/src/lib.rs (hunk 4 above)
Comment (RightFS) [reply pallets/ad/src/lib.rs:214]: 没看明白你在说什么

**Pull request merged**